GVFS/GVFS.Common/IProductUpgrader.cs

```diff
@@ -0,0 +1,35 @@
+using GVFS.Common.Git;
+using System;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Delegate to wrap install action steps in.
+    /// This can be used to report the beginning / end of each install step.
+    /// </summary>
+    /// <param name="method">The method to run inside wrapper</param>
+    /// <param name="message">The message to display</param>
+    /// <returns>success or failure return from the method run.</returns>
+    public delegate bool InstallActionWrapper(Func<bool> method, string message);
+
+    public interface IProductUpgrader
+    {
+        bool TryGetConfigAllowsUpgrade(out bool isConfigError, out string message);
+
+        bool TryInitialize(out string errorMessage);
+
+        bool TryGetNewerVersion(out Version newVersion, out string message);

```

Comment:
I get caught up on `GetNewerVersion` and am not sure if "get" means we are going to download it, or if we are just going to report the latest version.

Would something like `QueryNewestVersion` be more descriptive?

Reply:
agree. renamed to `TryQueryNewestVersion`